builder/builder.go

```diff
@@ -557,12 +554,30 @@ func (b *Builder) BuildUpdate(prefixflag string, minVersion int, format string,
 
 	// Create update metadata for the mix.
 	timer.Start("create update")
-	updatecmd := exec.Command(prefixflag+"swupd_create_update", "-S", b.Statedir, "--minversion", strconv.Itoa(minVersion), "-F", b.Format, "--osversion", b.Mixver)
-	updatecmd.Stdout = os.Stdout
-	updatecmd.Stderr = os.Stderr
-	err = updatecmd.Run()
-	if err != nil {
-		return errors.Wrapf(err, "failed to create update metadata")
+	if UseNewSwupdServer {
+		// TODO: move this to parsing configuration / parameter time.
+		if minVersion < 0 || minVersion > math.MaxUint32 {
+			return errors.Errorf("minVersion %d is out of range", minVersion)
+		}
+		// TODO: move this to parsing configuration / parameter time.
+		// TODO: should this be uint64?
```

Comment:
I don't think we'll max the format in our lifetime if it's a uint64.

Reply:
Format is "uint" with size not specified, the TODO was more to remember if we want to set the size of the number that holds format.